### PR TITLE
Fix linter errors in tests

### PR DESCRIPTION
**Pull request by henryjw**
A lot of files were modified, but the changes are mostly changing `var` declarations to `let` or `const`.

### CODE REVIEW

**Comment by tgriesser** [comment]
Cool, thanks @henryjw! One question, is `eslint-plugin-react` needed for anything or can that be removed?

**Comment by henryjw** [comment]
No problem.
For the packages, I had to install them because otherwise `eslint` errors out. Does liniting work fine for you without them?

**Comment by kibertoad** [comment]
@henryjw Why would you need react eslint for backend node.js library without any traces of React, though?..

**Comment by kibertoad** [comment]
@henryjw Linting used to work fine for me in the past, so I assume it is something on your local configuration end.

**Comment by henryjw** [comment]
@kibertoad you were right. It was something in my local setup. I've removed the additional dependencies.

File: package.json [hunk 1]
```diff
@@ -82,6 +82,7 @@
     "coveralls": "nyc report --reporter=text-lcov | coveralls",
     "dev": "rimraf ./lib && babel -w src --out-dir lib --copy-files",
     "lint": "eslint src/**",
+    "lint-test": "eslint test/",
```
Comment (kibertoad) [inline package.json]: shouldn't this be also included in travis file to be executed in CI?

**Comment by kibertoad** [comment]
@tgriesser @elhigu Do we want to merge this before https://github.com/tgriesser/knex/pull/2857 has landed? I assume that we really, really want that one to be completed and merged, and rebasing it against these changes is going to take some effort. Probably it's going to be easier to rebase this PR later on instead.

File: package.json (hunk 1 above)
Comment (tgriesser) [reply package.json]: The line above can just be change from 
```
"lint": "eslint src/**",
```
to
```
"lint": "eslint '{src,test}/**/*.js'",
```

**Comment by tgriesser** [comment]
> I assume that we really, really want that one to be completed and merged, and rebasing it against these changes is going to take some effort.

Yeah we definitely do want it to be merged, though that might take awhile and the lint errors in the tests always bug the hell out of me, I don't mind taking the time to merge this against those changes. 

Looking through the files changed in #2857 PR I actually don't think it'll be too hard, there's not too many test files changed and those that do don't really overlap too much with the lines changed by this PR.

@henryjw if you fix the line I mentioned above with the `lint` (remove `lint-test`), I'm good to merge this.

File: package.json (hunk 1 above)
Comment (henryjw) [reply package.json]: Fixed.

**Comment by henryjw** [comment]
@tgriesser I made the change but it looks like it broke the build. Does the build also take into account warnings? If so, then I'll have to resolve those as well.

**Comment by kibertoad** [comment]
@henryjw https://travis-ci.org/tgriesser/knex/jobs/441730589 -> judging by logs, those are pretty much errors.

**Comment by henryjw** [comment]
@kibertoad Yea that's strange. On my local they're warnings. Could it be that Travis treats warnings as errors?

**Comment by henryjw** [comment]
@tgriesser @kibertoad Pushed the fixes.

**Comment by tgriesser** [comment]
Great, thanks @henryjw!

**Pull request merged**